### PR TITLE
Add group by column stage

### PR DESCRIPTION
**Pull request by dagardner-nv**
## Description
* Adds new stage `GroupByColumnStage`

## By Submitting this PR I confirm:
- I am familiar with the [Contributing Guidelines](https://github.com/nv-morpheus/Morpheus/blob/main/docs/source/developer_guide/contributing.md).
- When the PR is ready for review, new or existing tests cover these changes.
- When the PR is ready for review, the documentation is up to date with these changes.

### CODE REVIEW

**Review by AnuradhaKaruppiah — COMMENTED** [review]
Is there an issue# for this change? I would like to understand why it is being done.

**Comment by dagardner-nv** [comment]
> Is there an issue# for this change? I would like to understand why it is being done.

The purpose of this stage is to ensure that the rest of the downstream stages in the pipeline get a unique message for each unique value in the column being grouped-by. 

The idea is that this would improve performance. In addition to this, if we know that each value in column 'A' has the same value, we can drop the column and store that value in the `ControlMessage.metadata` field.

I opened issue #1709 for this, but unfortunately I didn't create an issue prior to working on this feature.

File: morpheus/stages/preprocess/group_by_column_stage.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        return False
+
+    def on_data(self, message: MessageMeta) -> list[MessageMeta]:
+        with message.mutable_dataframe() as df:
+            grouper = df.groupby(self._column_name)
+
+        ouptut_messages = []
+        for group_name in grouper.groups.keys():
+            group_df = grouper.get_group(group_name)
+            ouptut_messages.append(MessageMeta(group_df))
```
Comment (AnuradhaKaruppiah) [inline morpheus/stages/preprocess/group_by_column_stage.py:58]: pls fix the spelling here.

File: morpheus/stages/preprocess/group_by_column_stage.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        self._column_name = column_name
+
+    @property
+    def name(self) -> str:
+        return "group-by-column"
+
+    def accepted_types(self) -> tuple:
+        return (MessageMeta, )
+
+    def supports_cpp_node(self) -> bool:
```
Comment (AnuradhaKaruppiah) [inline morpheus/stages/preprocess/group_by_column_stage.py:48]: @method?

File: morpheus/stages/preprocess/group_by_column_stage.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            grouper = df.groupby(self._column_name)
+
+        ouptut_messages = []
+        for group_name in grouper.groups.keys():
+            group_df = grouper.get_group(group_name)
+            ouptut_messages.append(MessageMeta(group_df))
+
+        return ouptut_messages
+
+    def _build_single(self, builder: mrc.Builder, input_node: mrc.SegmentObject) -> mrc.SegmentObject:
```
Comment (AnuradhaKaruppiah) [inline morpheus/stages/preprocess/group_by_column_stage.py:62]: Where is this private method being used?

**Comment by AnuradhaKaruppiah** [comment]
LGTM, some minor suggestions.

File: morpheus/stages/preprocess/group_by_column_stage.py (hunk 1 above)
Comment (dagardner-nv) [reply morpheus/stages/preprocess/group_by_column_stage.py:58]: Good catch, thanks!

File: morpheus/stages/preprocess/group_by_column_stage.py (hunk 3 above)
Comment (dagardner-nv) [reply morpheus/stages/preprocess/group_by_column_stage.py:76]: The parent class `SinglePortStage` defines this, and will be invoked by the Pipeline (via `StageBase.build`) during the build phase.

**Review by AnuradhaKaruppiah — APPROVED** [review]

**Comment by dagardner-nv** [comment]
test

**Review by AnuradhaKaruppiah — APPROVED** [review]

**Review by AnuradhaKaruppiah — APPROVED** [review]

**Pull request merged**